### PR TITLE
Stat tracker

### PR DESCRIPTION
**Pull request by Sgalvin36**
Method: from_csv(info)
Function: runs helper methods for each class factory and stores all the data aggregated together

Method game_factory(info)
Function: runs a loop over the games csv creating a game object for each row

Method team_factory(info)
Function: runs a loop over the teams csv creating a team object for each row

Method game_teams_factory(info)
Function: runs a loop over the game_teams csv creating a game_team object for each row

Created a test suite for each method in isolation and combined together.

### CODE REVIEW

**Comment by stefanjshepard** [comment]
Note the conversion methods in the rspec testing suites for game/ teams/ game teams.

**Comment by jimmacur** [comment]
This is looking great. We are possibly going to do some refactoring if there is time later in the project. For now it functions well and everything follows SRP.

**Review by jimmacur — COMMENTED** [review]
YES!

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        data_collect = StatTracker.new
+        data_collect.games.replace(StatTracker.game_factory(info))
+        data_collect.teams.replace(StatTracker.team_factory(info))
+        data_collect.game_teams.replace(StatTracker.game_team_factory(info))
+        data_collect
+    end
+
+    def self.game_factory(info)
+        games = []
+        CSV.foreach(info[:games], headers: true, header_converters: :symbol) do |row|
```
Comment (CCirbo) [inline lib/stat_tracker.rb:25]: and here too.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
+require 'CSV'
+require_relative './game'
+require_relative './team'
+require_relative './game_team'
+
+class StatTracker
+    attr_reader :games, :teams, :game_teams
+
+    def initialize
+        @games = []
```
Comment (CCirbo) [inline lib/stat_tracker.rb:10]: I have some ideas for refactoring for later if you want here.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        @teams = []
+        @game_teams = []
+    end
+
+    def self.from_csv(info)
+        data_collect = StatTracker.new
```
Comment (CCirbo) [inline lib/stat_tracker.rb:16]: here too.

**Review by CCirbo — COMMENTED** [review]
Yay, this looks amazing, so much work and so glad it is all working!

**Pull request merged**